### PR TITLE
feat(HDRReader): add vtkHDRReader

### LINKED ISSUE #3145: [Feature] Support for HDR images
### Motivation

Support for the Radiance HDR (RGBE) image format for environment maps

### Detailed Description

Add a new vtkHDRReader

```js
const reader = vtkHDRReader.newInstance();
reader.parse(fileReader.result);
const imageData = reader.getOutputData();
```

### PR DESCRIPTION
**Pull request by daker**
### Context
This PR adds support for loading Radiance HDR (RGBE) image format for environment maps

Fixes #3145

### Results
image

### Changes
- [x] Documentation and TypeScript definitions were updated to match those changes

### PR and Code Checklist
- [x] [semantic-release](https://github.com/semantic-release/semantic-release) commit messages
- [x] Run `npm run reformat` to have correctly formatted code

### Testing
- [ ] This change adds or fixes unit tests 
- [ ] Tested environment:
  - **vtk.js**: 
  - **OS**: 
  - **Browser**:

### CODE REVIEW

File: Sources/IO/Image/HDRReader/index.js [hunk 1]
```diff
@@ -0,0 +1,437 @@
...
+    imageData.getPointData().setScalars(dataArray);
+    model.output[0] = imageData;
+  };
+
+  publicAPI.requestData = (inData, outData) => {
+    publicAPI.parse(model.parseData);
+  };
+
+  publicAPI.getWidth = () => model.dataExtent[1] - model.dataExtent[0] + 1;
+  publicAPI.getHeight = () => model.dataExtent[3] - model.dataExtent[2] + 1;
```
Comment (finetjul) [inline Sources/IO/Image/HDRReader/index.js:381]: maybe `return model.output[0].getDimensions()[0]` (with some failsafe ?)

File: Sources/IO/Image/HDRReader/index.js [hunk 2]
```diff
@@ -0,0 +1,437 @@
...
+
+    model.dataExtent[0] = 0;
+    model.dataExtent[1] = header.width - 1;
+    model.dataExtent[2] = 0;
+    model.dataExtent[3] = header.height - 1;
+
+    model.dataSpacing[0] = 1;
+    model.dataSpacing[1] = header.pixelAspect;
+    model.dataSpacing[1] = 1;
+    model.dataSpacing[2] = 1;
```
Comment (finetjul) [inline Sources/IO/Image/HDRReader/index.js:358]: do not store in reader, all the information should be in the image data

File: Sources/IO/Image/HDRReader/index.js [hunk 3]
```diff
@@ -0,0 +1,437 @@
...
+
+    const imageData = vtkImageData.newInstance();
+    imageData.setDimensions(header.width, header.height, 1);
+    imageData.setExtent(model.dataExtent);
+    imageData.setSpacing(model.dataSpacing);
+    imageData.setOrigin(0, 0, 0);
```
Comment (finetjul) [inline Sources/IO/Image/HDRReader/index.js:364]: isn't origin already at 0,0,0 ?

File: Sources/IO/Image/HDRReader/index.js [hunk 4]
```diff
@@ -0,0 +1,437 @@
...
+}
+
+// ----------------------------------------------------------------------------
+// Object factory
+// ----------------------------------------------------------------------------
+
+const DEFAULT_VALUES = {
+  gamma: 1.0,
+  exposure: 1.0,
+  pixelAspect: 1.0,
```
Comment (finetjul) [inline Sources/IO/Image/HDRReader/index.js:391]: does not seem to be used

File: Sources/IO/Image/HDRReader/example/index.js [hunk 5]
```diff
@@ -0,0 +1,122 @@
+import '@kitware/vtk.js/favicon';
```
[the remark this answers is not in the document]
Comment (daker) [reply Sources/IO/Image/HDRReader/example/index.js:1]: Can you share the image template with the icon ?

[the remark this answers is not in the document]
Comment (finetjul) [reply Sources/IO/Image/HDRReader/example/index.js:1]: [File VTK.js icons.pptx](https://github.com/user-attachments/files/17373563/File.VTK.js.icons.pptx)
Here you go

File: Sources/IO/Image/HDRReader/Utils.js [hunk 6]
```diff
@@ -0,0 +1,43 @@
+import * as vtkMath from '@kitware/vtk.js/Common/Core/Math';
```
Comment (finetjul) [inline Sources/IO/Image/HDRReader/Utils.js:1]: `import * as vtkMath from 'vtk.js/Sources/Common/Core/Math';`

File: Sources/IO/Image/HDRReader/index.js [hunk 7]
```diff
@@ -0,0 +1,421 @@
...
+
+// Enable data soure for DataAccessHelper
+import 'vtk.js/Sources/IO/Core/DataAccessHelper/LiteHttpDataAccessHelper'; // Just need HTTP
+// import 'vtk.js/Sources/IO/Core/DataAccessHelper/HttpDataAccessHelper'; // HTTP + zip
+// import 'vtk.js/Sources/IO/Core/DataAccessHelper/HtmlDataAccessHelper'; // html + base64 + zip
+// import 'vtk.js/Sources/IO/Core/DataAccessHelper/JSZipDataAccessHelper'; // zip
+
+import macro from 'vtk.js/Sources/macros';
+import DataAccessHelper from 'vtk.js/Sources/IO/Core/DataAccessHelper';
+import * as vtkMath from '@kitware/vtk.js/Common/Core/Math';
```
Comment (finetjul) [inline Sources/IO/Image/HDRReader/index.js:11]: import * as vtkMath from 'vtk.js/Sources/Common/Core/Math';

**Review by finetjul — CHANGES_REQUESTED** [review]

**Review by finetjul — APPROVED** [review]
LGTM

**Comment by daker** [comment]
@floryst i rebased the branch

**Comment by floryst** [comment]
Thanks!

**Pull request merged**